.envrc

```diff
@@ -0,0 +1 @@
+use flake .
```

Comment:
this is just for direnv stuff right? any reason why this needs to be committed to the repo? Or should we add `.envrc` to gitignore so that people can use `.envrc` as they choose

Reply:
I left this here since I am unaware of what best practice is for direnv.
Adding this allows for direnv to automatically enter the nix development environment.
Even if we keep this file, the user should still need to run `direnv allow .` once  before direnv actually does anything, but anyone who wants to use it can choose to allow direnv and automatically use the nix flake upon entering the repo dir. 
There is potentially useful tooling around direnv that can take advantage of this such as direnv-mode for emacs, and I am sure other editors have their equivalents.
I think this would be a good compromise since those who don't use it would be unaffected and those who want to can easily opt in. 
If we add the file then, we should probably add `.direnv/` to the `.gitignore`.